httpcore/_async/connection_pool.py

```diff
@@ -147,6 +147,24 @@ def _connection_acquiry_lock(self) -> AsyncLock:
             self._internal_connection_acquiry_lock = self._backend.create_lock()
         return self._internal_connection_acquiry_lock
 
+    def _create_connection(
+        self,
+        origin: Tuple[bytes, bytes, int],
+        backend: AsyncBackend,
+        http2: bool = False,
+        uds: str = None,
+        ssl_context: SSLContext = None,
+        local_address: str = None,
+    ) -> AsyncHTTPConnection:
+        return AsyncHTTPConnection(
+            origin=origin,
+            http2=self._http2,
```

Comment:
Don't we want to use the arguments instead of the instance attributes?

```suggestion
            http2=http2,
```

Reply:
Ah good catch. Prompted a nice clean-up... Don't pass the arguments through at all.

We really only need to be passing the `origin` argument. Everything else is available as attributes on the class.